nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/document/ExtractDocumentText.java

```diff
@@ -0,0 +1,130 @@
+        "and Microsoft Office files.")
+public class ExtractDocumentText extends AbstractProcessor {
+    private static final String TEXT_PLAIN = "text/plain";
+
+    public static final String FIELD_MAX_TEXT_LENGTH = "max-text-length";
+    public static final String FIELD_SUCCESS = "success";
+    public static final String FIELD_FAILURE = "failure";
+
+    public static final PropertyDescriptor MAX_TEXT_LENGTH = new PropertyDescriptor.Builder()
+            .name(FIELD_MAX_TEXT_LENGTH)
+            .displayName("Max Output Text Length")
+            .description("The maximum length of text to retrieve. This is used to limit memory usage for " +
+                    "dealing with large files. Specify -1 for unlimited length.")
+            .required(false).defaultValue("-1").addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES).build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name(FIELD_SUCCESS)
+            .description("Content extraction successful").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name(FIELD_FAILURE)
+            .description("Content extraction failed").build();
+
+    private List<PropertyDescriptor> descriptors = Collections.unmodifiableList(Arrays.asList(MAX_TEXT_LENGTH));
+    private Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final int maxTextLength = context.getProperty(MAX_TEXT_LENGTH).evaluateAttributeExpressions(flowFile).asInteger();
+        final String filename = flowFile.getAttribute("filename");
+
+        try {
+            final AtomicReference<String> type = new AtomicReference<>();
+            final AtomicReference<Boolean> exceptionThrown = new AtomicReference<>(false);
+
+            flowFile = session.write(flowFile, (inputStream, outputStream) -> {
+                BufferedInputStream buffStream = new BufferedInputStream(inputStream);
+                Tika tika = new Tika();
+                String text = "";
+                try {
+                    type.set(tika.detect(buffStream, filename));
+                    tika.setMaxStringLength(maxTextLength);
+                    text = tika.parseToString(buffStream);
+
+                } catch (TikaException e) {
+                    getLogger().error("Parsing failed ", e);
```

Comment:
This log should be removed as the outer exception handling should indicate the failure details.
```suggestion
```